### PR TITLE
Upgrade ckan 2.10

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

Upgrade of CKAN to 2.10, this will be merged into main-2.10 after approval as the existing govuk stack is still using 2.9.3 and won't be able to support 2.10 until CKAN is replatformed.

## Reference

https://trello.com/c/bEsbEk1m/1151-run-security-check-and-resolve-critical-issues

### CODE REVIEW

File: ckan.ini [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+ckan.auth.enable_cookie_auth_in_api = true
+ckan.auth.route_after_login = dashboard.datasets
+
+## User Account Creation Setting
+; ckan.valid_email_regexes = .gov.uk$ .nhs.uk$ .nhs.net$ .ac.uk$ .os.uk$ .mod.uk$ .police.uk$ .bl.uk$
+
+## CSRF Protection #############################################################
+WTF_CSRF_ENABLED = true
+WTF_CSRF_CHECK_DEFAULT = true
+WTF_CSRF_SECRET_KEY = _e6qkoYldyll9U3zfy5n6rg-JxCPY4DUfy5LHu3qnUo
```
Comment (sengi) [inline ckan.ini:114]: Is this used for anything real? If so you probably need to change it 😅

**Review by sengi — APPROVED** [review]
All looks reasonable to me, apart from the suspicious-looking secret string in the config file that you might need to change (unless it's unused, in which case I'd remove it or change it to something that's obviously a test/dummy value, whatever makes sense for your use case)

File: ckan.ini (hunk 1 above)
Comment (kentsanggds) [reply ckan.ini:114]: Not used for anything real, just auto generated so should be ok for local dev use, I'll make sure that the ckan chart auto generates when deployed

**Comment by kentsanggds** [comment]
Closing this in favour of https://github.com/alphagov/ckanext-datagovuk/pull/1662

**Pull request closed without merging**